File: python/kindergarten-garden/kindergarten_garden.py
default_students = ["Alice", "Bob", "Charlie", "David", "Eve", "Fred", "Ginny", "Harriet", "Ileana", "Joseph", "Kincaid", "Larry"]
plant_map = {
    "C": "Clover",
    "G": "Grass",
    "R": "Radishes",
    "V": "Violets",
}

class Garden:
    def __init__(self, diagram, students=default_students):
        self.students = sorted(students)
        self.rows = [list(line) for line in diagram.split("\n")]

    def plants(self, name):
        index = self.students.index(name) * 2
        plants = []
        for r in self.rows:
            plants.extend([plant_map[plant] for plant in r[index:index+2]])
        return plants

File: ruby/kindergarten-garden/kindergarten_garden.rb
class Garden
  STUDENTS = %i[alice bob charlie david eve fred ginny harriet ileana joseph kincaid larry]
  PLANTS = {
    :V => :violets,
    :C => :clover,
    :G => :grass,
    :R => :radishes,
  }

  def initialize(garden)
    @garden = garden.split("\n")
  end

  def method_missing(method)
    start = STUDENTS.index(method) * 2
    (0..1).each_with_object([]) { |row, out|
      (start..(start + 1)).each { |col|
        out << PLANTS[@garden[row][col].to_sym]
      }
    }
  end
end
